src/__test__/components/plots-and-tables/markerHeatmap.test.jsx

```diff
@@ -0,0 +1,235 @@
+      },
+      'louvain-1': {
+        name: 'Cluster 1',
+        color: '#8c564b',
+        cellIds: new Set([5, 6, 7, 8]),
+        rootNode: false,
+        type: 'cellSets',
+      },
+      'louvain-2': {
+        name: 'Cluster 2',
+        color: '#d62728',
+        cellIds: new Set([9, 10, 11, 12]),
+        rootNode: false,
+        type: 'cellSets',
+      },
+      'louvain-3': {
+        name: 'Cluster 3',
+        color: '#2ca02c',
+        cellIds: new Set([13, 14, 15, 16]),
+        rootNode: false,
+        type: 'cellSets',
+      },
+      views: [],
+    },
+    hidden: 'Set()',
+  },
+  genes: {
+    markers: {
+      error: false,
+    },
+    expression: {
+      loading: [],
+
+      data: {
+        gene0: {
+          rawExpression: { expression: [5, 5, 5, 5, 0, 4, 4, 4, 2, 3, 4, 5, 0, 0, 4, 5] },
+        },
+        gene1: {
+          rawExpression: { expression: [1, 2, 3, 4, 5, 4, 5, 5, 3, 0, 0, 0, 0, 0, 0, 0] },
+        },
+        gene3: {
+          rawExpression: { expression: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 0, 1, 0] },
+        },
+      },
+      views: {
+        [plotUuid]: {
+          data: ['gene0', 'gene1', 'gene3'],
+        },
+      },
+    },
+    properties: { views: { [plotUuid]: [] } },
+  },
+};
+
+const experimentId = 'randomExperiment';
+let store = null;
+let loadConfigSpy = null;
+let loadMarkersSpy;
+let configUpdatedSpy;
+let fetchCachedWorkSpy;
+
+const renderHeatmapPage = async () => {
+  store = createStore(rootReducer, _.cloneDeep(defaultStore), applyMiddleware(thunk));
+
+  rtl.render(
+    <Provider store={store}>
+      <MarkerHeatmap
+        experimentId={experimentId}
+      />
+    </Provider>,
+  );
+  await rtl.waitFor(() => expect(loadConfigSpy).toHaveBeenCalledTimes(1));
+};
+
+describe('Marker heatmap plot', () => {
+  beforeAll(async () => {
+    await preloadAll();
+  });
+  beforeEach(() => {
+    jest.clearAllMocks();
+    fetchMock.resetMocks();
+    fetchMock.doMock();
+    fetchMock.mockResponse(JSON.stringify({}), { status: 404, statusText: '404 Not Found' });
+    loadConfigSpy = jest.spyOn(loadConfig, 'default');
+    loadMarkersSpy = jest.spyOn(markerGenesLoaded, 'default');
+    configUpdatedSpy = jest.spyOn(configUpdated, 'default');
+  });
+
+  // marker heatmap renders
+  it('loads initially', async () => {
+    await renderHeatmapPage();
+
+    expect(loadMarkersSpy).toHaveBeenCalled();
+  });
+
+  it('loads marker genes on selecting', async () => {
+    await renderHeatmapPage();
+
+    const geneSelection = rtl.screen.getByText('Gene selection');
+    userEvent.click(geneSelection);
+    const nGenesInput = rtl.getByTestId(geneSelection.parentElement, 'num-genes');
+    userEvent.type(nGenesInput, 10);
+
+    const runButton = rtl.getByText(geneSelection.parentElement, 'Run');
+    userEvent.click(runButton);
+
+    expect(loadMarkersSpy).toHaveBeenCalled();
+  });
+
+  it('sorts genes properly when adding a gene', async () => {
```

Comment:
also missing tests for the heatmap states: that when we are loading a gene, the heatmap is in loading state. When there's been an error in loading a gene, the heatmap is in error state.

Reply:
Do we  not have tests for this - loadGeneExpression.test.jsx do i  need to test for that here again?

Reply:
in loadGeneExpression we test the store, in here you would test if the component correctly responds to the updates of the store. So you would check that `Skeleton` gets rendered when the relevant parts of the redux store are in loading state. What you can do is to mock the store, settings some of the contents to be loading: true (like cellSets or geneExpression for this plotUuid) and then assert that instead of MarkerHeatmap, we render `Skeleton`.